pkg/crd/persistentip/v1alpha1/types.go

```diff
@@ -0,0 +1,46 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//go:generate go run sigs.k8s.io/controller-tools/cmd/controller-gen@v0.13.0 paths=./... object output:dir=./
+
+//go:generate go run k8s.io/code-generator/cmd/client-gen@v0.28.0 client-gen --go-header-file ../../../../hack/custom-boilerplate.go.txt --clientset-name versioned --input-base "" --input github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1 --output-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/clientset ..
+
+//go:generate go run k8s.io/code-generator/cmd/lister-gen@v0.28.0 lister-gen --go-header-file ../../../../hack/custom-boilerplate.go.txt --input-dirs github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1 --output-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/listers ..
+
+//go:generate go run k8s.io/code-generator/cmd/informer-gen@v0.28.0 informer-gen --go-header-file ../../../../hack/custom-boilerplate.go.txt --input-dirs github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1 --versioned-clientset-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/clientset/versioned --listers-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/listers --output-package github.com/maiqueb/persistentips/pkg/crd/persistentip/v1alpha1/apis/informers ..
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=persistentipamallocations,shortName=pip,singular=persistentipamallocation,scope=Namespaced
+// +kubebuilder:storageversion
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PersistentIPAMAllocation is the Schema for the PersistentIPAMAllocation API
+type PersistentIPAMAllocation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec PersistentIPAMAllocationSpec `json:"spec,omitempty"`
+}
+
+type PersistentIPAMAllocationSpec struct {
+	// The network attachment definition name for which this persistent allocation was created
+	Network string `json:"network"`
+	// +optional
+	// The pod interface name for which this allocation was created
+	Interface string `json:"interface,omitempty"`
```

Comment:
Are we sure this has to be optional ?

Reply:
Right, it shouldn't be.

My train of thought was for workloads with a single interface, you don't need to specify it, but it makes a clunky flow.

Reply:
Done.

Removing this had no impact in the generated code.

I wonder if I should actually remove the `omitempty` option as well.

Reply:
Sticking w/ optional for now, we can always remove that later on once we get feedback from k8snetworkplumbingwg.